common/src/Model/Node.h

```diff
@@ -306,193 +307,106 @@ namespace TrenchBroom {
             void clearIssues() const;
         public: // visitors
             /**
-             * Visit this node, and recursively visit its children
-             */
-            template <class V>
-            void acceptAndRecurse(V& visitor) {
-                accept(visitor);
-                if (!visitor.recursionStopped() && !visitor.cancelled())
-                    recurse(visitor);
-            }
-
-            /**
-             * Visit this node, and recursively visit its children
-             */
-            template <class V>
-            void acceptAndRecurse(V& visitor) const {
-                accept(visitor);
-                if (!visitor.recursionStopped() && !visitor.cancelled())
-                    recurse(visitor);
-            }
-
-            /**
-             * For each node in the given range, visit it, and then recursively visit its children,
-             * until the visitor gets cancelled
-             */
-            template <typename I, typename V>
-            static void acceptAndRecurse(I cur, I end, V& visitor) {
-                while (cur != end && !visitor.cancelled()) {
-                    (*cur)->acceptAndRecurse(visitor);
-                    ++cur;
-                }
-            }
-
-            /**
-             * Visit this node, and recursively visit its parents
-             */
-            template <class V>
-            void acceptAndEscalate(V& visitor) {
-                accept(visitor);
-                if (!visitor.recursionStopped() && !visitor.cancelled())
-                    escalate(visitor);
-            }
-
-            /**
-             * Visit this node, and recursively visit its parents
-             */
-            template <class V>
-            void acceptAndEscalate(V& visitor) const {
-                accept(visitor);
-                if (!visitor.recursionStopped() && !visitor.cancelled())
-                    escalate(visitor);
-            }
-
-            /**
-             * For each node in the given range, visit it, and then recursively visit its parents,
-             * until the visitor gets cancelled
-             */
-            template <typename I, typename V>
-            static void acceptAndEscalate(I cur, I end, V& visitor) {
-                while (cur != end && !visitor.cancelled()) {
-                    (*cur)->acceptAndEscalate(visitor);
-                    ++cur;
-                }
-            }
-
-            /**
-             * Visit this node (not visiting parents or children)
+             * Visit this node with the given lambda and return the lambda's return value or nothing
+             * if the lambda doesn't return anything.
+             *
+             * Passes a non-const pointer to this node to the lambda.
              */
-            template <class V>
-            void accept(V& visitor) {
+            template <typename L>
+            auto accept(const L& lambda) {
```

Comment:
Yeah, this naming conflict is unfortunate. But "visit" is wrong(er) than "accept", because it's the visitee who accepts the visitor. I'd rather rename the helper methods to something else, but I couldn't think of a better name.